dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/services/jobService/data/PutJobData.kt

```diff
@@ -96,18 +104,22 @@ data class PutJobData(private val items: List<Pair<String, Path>>,
     private fun dataToDs3Objects(item: Pair<String, Path>): Iterable<Ds3Object> {
         val localDelim = item.second.fileSystem.separator
         val parent = item.second.parent
-        val ds3ObjectList = Files.walk(item.second)
-                .filter { !(Files.isDirectory(it) && Files.list(it).findAny().isPresent) }
-                .map {
-                    Ds3Object(if (Files.isDirectory(it)) {
-                        targetDir + parent.relativize(it).toString().replace(localDelim, "/") + "/"
-                    } else {
-                        targetDir + parent.relativize(it).toString().replace(localDelim, "/")
+        val paths = Files.walk(item.second).use { stream: Stream<Path> ->
+            stream.iterator().asSequence()
+                    .takeWhile { !cancelled!!.invoke() }
+                    .filter { !(Files.isDirectory(it) && Files.list(it).use { f -> f.findAny().isPresent }) }
+                    .map {
+                        Ds3Object(if (Files.isDirectory(it)) {
+                            targetDir + parent.relativize(it).toString().replace(localDelim, "/") + "/"
+                        } else {
+                            targetDir + parent.relativize(it).toString().replace(localDelim, "/")
+                        }
+                                , if (Files.isDirectory(it)) 0L else Files.size(it))
                     }
-                            , if (Files.isDirectory(it)) 0L else Files.size(it))
-                }.collect(GuavaCollectors.immutableList())
-        ds3ObjectList.forEach({ prefixMap.put(it.name, item.second) })
-        return ds3ObjectList
+                    .toList()
+        }
+        paths.forEach { prefixMap.put(it.name, item.second) }
+        return paths.asIterable()
```

Comment:
Isn't a list already an iterable?  Why this call?

Reply:
Good catch.